metric_learn/itml.py

```diff
@@ -138,3 +135,47 @@ def _vector_norm(X):
 else:
   def _vector_norm(X):
     return np.linalg.norm(X, axis=1)
+
+
+class ITML_Supervised(ITML):
+  """Information Theoretic Metric Learning (ITML)"""
+  def __init__(self, gamma=1., max_iters=1000, convergence_threshold=1e-3, num_constraints=None,
+    bounds=None, A0=None, verbose=False):
+    """Initialize the learner.
+
+    Parameters
+    ----------
+    gamma : float, optional
+        value for slack variables
+    max_iters : int, optional
+    convergence_threshold : float, optional
+    num_constraints: int, needed for .fit()
+    verbose : bool, optional
+        if True, prints information while learning
+    """
+    self.params = {
```

Comment:
I'd prefer to call the base constructor here:

    ITML.__init__(self, gamma=gamma, max_iters=max_iters, convergence_threshold=convergence_threshold,
                  bounds=bounds, A0=A0, verbose=verbose)
    self.params['num_constraints'] = num_constraints